cpp/src/arrow/util/thread-pool.cc

```diff
@@ -294,9 +294,14 @@ std::shared_ptr<ThreadPool> ThreadPool::MakeCpuThreadPool() {
   return pool;
 }
 
+ThreadPool* arrow_global_thread_pool = nullptr;
+
 ThreadPool* GetCpuThreadPool() {
-  static std::shared_ptr<ThreadPool> singleton = ThreadPool::MakeCpuThreadPool();
-  return singleton.get();
+  if (!arrow_global_thread_pool) {
```

Comment:
This is not thread-safe. There is a reason why I used a static singleton, and that reason is thread safety.

Reply:
I think we can even demote it to a unique_ptr, since the object is never re-assigned.

Reply:
Yeah, you are both right (this was just to try out if removing the static shared_ptr makes a difference)